Source/spelldat.h

```diff
@@ -203,24 +205,56 @@ enum class MissileID : int8_t {
 	// clang-format on
 };
 
+enum class SpellDataFlags : uint8_t {
+	// The lower 2 bytes are used to store MagicType.
+	Fire = static_cast<uint8_t>(MagicType::Fire),
+	Lightning = static_cast<uint8_t>(MagicType::Lightning),
+	Magic = static_cast<uint8_t>(MagicType::Magic),
+	Targeted = 1U << 2,
+	AllowedInTown = 1U << 3,
+};
+use_enum_as_flags(SpellDataFlags);
+
 struct SpellData {
-	SpellID sName;
-	uint8_t sManaCost;
-	MagicType sType;
 	const char *sNameText;
+	_sfx_id sSFX;
+	uint16_t bookCost100;
+	uint8_t staffCost10;
+	uint8_t sManaCost;
+	SpellDataFlags flags;
 	int8_t sBookLvl;
 	int8_t sStaffLvl;
-	bool sTargeted;
-	bool sTownSpell;
-	int16_t sMinInt;
-	_sfx_id sSFX;
-	MissileID sMissiles[3];
+	uint8_t minInt;
+	MissileID sMissiles[2];
 	uint8_t sManaAdj;
 	uint8_t sMinMana;
 	uint8_t sStaffMin;
 	uint8_t sStaffMax;
-	uint32_t sBookCost;
-	uint16_t sStaffCost;
+
+	[[nodiscard]] MagicType type() const
+	{
+		return static_cast<MagicType>(static_cast<std::underlying_type<SpellDataFlags>::type>(flags) & 0b11U);
+	}
+
+	[[nodiscard]] uint32_t bookCost() const
+	{
+		return bookCost100 * 100;
+	}
+
+	[[nodiscard]] uint16_t staffCost() const
+	{
+		return staffCost10 * 10;

```

Comment:
this feels very agressive

Reply:
I think it makes sense. Why use a 32-bit number so you can store a bunch of zeroes? As long as it's not changing the cost of things from what they were originally, I don't see the issue.

Reply:
imo it's harder to read and modefy

Reply:
True but the struct is exactly 24 bytes. Adding even a single byte to this struct will increase its size to 32 bytes (on 64-bit systems) due to alignment requirements for the `sNameText` pointer.

Reply:
We could change `bookCost100` to `bookCost10` for free though.

Reply:
Are we memory strapped on 64bit systems?

Reply:
Alignment requirements are still there on 32-bit systems.
On a 32-bit system, the struct is currently exactly 20 bytes; adding even a single a byte will increase the size to 24 bytes.